crypto/sessionkey.go

```diff
@@ -273,45 +318,24 @@ func (sk *SessionKey) DecryptAndVerify(dataPacket []byte, verifyKeyRing *KeyRing
 	default:
 		return nil, errors.New("gopenpgp: invalid packet type")
 	}
-	_, err = decBuf.ReadFrom(decrypted)
-	if err != nil {
-		return nil, errors.Wrap(err, "gopenpgp: unable to read from decrypted symmetric packet")
-	}
 
 	config := &packet.Config{
 		Time: getTimeGenerator(),
 	}
 
 	// Push decrypted packet as literal packet and use openpgp's reader
-
 	if verifyKeyRing != nil {
 		keyring = verifyKeyRing.entities
 	} else {
 		keyring = openpgp.EntityList{}
 	}
 
-	md, err := openpgp.ReadMessage(&decBuf, keyring, nil, config)
+	md, err := openpgp.ReadMessage(decrypted, keyring, nil, config)
 	if err != nil {
 		return nil, errors.Wrap(err, "gopenpgp: unable to decode symmetric packet")
 	}
 
-	messageBuf := new(bytes.Buffer)
-	_, err = messageBuf.ReadFrom(md.UnverifiedBody)
-	if err != nil {
-		return nil, errors.Wrap(err, "gopenpgp: error in reading message body")
-	}
-
-	if verifyKeyRing != nil {
-		processSignatureExpiration(md, verifyTime)
-		err = verifyDetailsSignature(md, verifyKeyRing)
-	}
-
-	return &PlainMessage{
-		Data:     messageBuf.Bytes(),
-		TextType: !md.LiteralData.IsBinary,
-		Filename: md.LiteralData.FileName,
-		Time:     md.LiteralData.Time,
-	}, err
+	return md, err
```

Comment:
```suggestion
	return md, nil
```

I like ensuring here the error's `nil` and we wrap the error